util/test/start_test.py

```diff
@@ -122,6 +122,9 @@ def run_tests(tests):
     # print out Chapel environment
     print_chapel_environment()
 
+    # look for and execute PRETEST's
+    check_for_pretest()
```

Comment:
Hmmm... I would think that putting this here would cause the PRETESTS to be run multiple times in a paratest run. What makes more sense to me would be to modify `sub_test` to do this operation on a directory right before testing that directory.

Reply:
> It generates a number of sub directories, which sub_test doesn't recurse into by default (or at all?).

This is all based on ancient memories and guesses:

I think that's because our test scripts look at all the test directories' structure before doing anything (?).  I'm fairly certain this is true of paratest anyway... less certain about start_test.  My suggestion would be to have the directories be under git control (e.g., put some breadcrumb into them to make sure they exist within git — this would also mean removing the .gitignore, I think?)) and I think that should be sufficient to have the scripts descend into them even if there are no tests in them until your script runs (?).

Reply:
This is more or less what I ended up doing. Each of the sub directories now contains it's own `PRETEST` so they can be individually run. I tweaked `start_test` and `paratest` to add test directories that contain `PRETEST` (not just `.chpl` and the like). This works fine for `start_test`, because it just runs in order. But for `paratest` it is allocating one node to handle all 1920 tests (because at first pass it only looks like 20ish files, but after `PRETEST` there are many more). I think a possibility for this is to make `PRETEST` a two level process for `paratest` (execute, then check for source files), but I am still thinking about the best way to do this

Reply:
It might well be worthwhile to try to get the cases that are expected to compile into one file to reduce the number of tests / testing time

Reply:
I did change the cases that are expected to compile into one file, which eliminates about 300 files. I also found that using a `NOTEST` instead of `.notest` for `generate-tests.chpl` "tricks" `paratest` into giving each of the sub dirs to different nodes, which significantly speeds up execution (each node now only handles max 192 files)